From efadf4f4b02ebce35872c17f49c809edf99a4fa3 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Sun, 8 Aug 2021 12:13:16 +0800
Subject: [PATCH 10/12] opcodes: LoongArch: add "ret" instruction to reduce
 typing

2021-10-25  WANG Xuerui <git@xen0n.name>

opcodes/
        * loongarch-opc.c: Add "ret" to loongarch_jmp_opcodes.
---
 opcodes/loongarch-opc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/opcodes/loongarch-opc.c b/opcodes/loongarch-opc.c
index 1f8f9227a1f..687bfd776c5 100644
--- a/opcodes/loongarch-opc.c
+++ b/opcodes/loongarch-opc.c
@@ -770,6 +770,7 @@ static struct loongarch_opcode loongarch_jmp_opcodes[] =
   { 0x0,	0x0,		"bltu",		"r,r,la",			"bgtu %2,%1,%%pcrel(%3)",	0, 0, 0 },
   { 0x0,	0x0,		"bltz",		"r,la",				"bgt $r0,%1,%%pcrel(%2)",	0, 0, 0 },
   { 0x0,	0x0,		"jr",		"r",				"jirl $r0,%1,0",		0, 0, 0 },
+  { 0x0,	0x0,		"ret",		"",				"jirl $r0,$r1,0",		0, 0, 0 },
 
   { 0 } /* Terminate the list.  */
 };
-- 
2.33.1

